include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -842,6 +843,44 @@ class __static_monotonic_dispatcher<::std::integer_sequence<::std::uint16_t, _X,
     }
 };
 
+// This exception handler is intended to handle a software workaround by IGC for a hardware bug that
+// causes IGC to throw an exception for certain integrated graphics devices with -O0 compilation and
+// a required sub-group size of 32.
+struct __bypass_sycl_kernel_not_supported
+{
+    void
+    operator()(const sycl::exception& __e) const
+    {
+        // The SYCL spec compliant solution would be to compare __e.code() and sycl::errc::kernel_not_supported
+        // and rethrow the encountered exception if the two do not compare equal. However, the icpx compiler currently
+        // returns a generic error code in violation of the SYCL spec which has a value of 7. If we are using the Intel
+        // compiler, then compare the value of the error code. Otherwise, assume the implementation is spec compliant.
+#if _ONEDPL_SYCL_KERNEL_NOT_SUPPORTED_EXCEPTION_BROKEN
+        if (__e.code().value() != 7)
+            throw;
```

Comment:
One more question which is not trivial to me. Should we rethrow that exception? oneDPL algorithm selection causes it by using an optional feature, handles it and choses a backup option. I see no value by passing it to the user, but I am not very experienced here, so I would seek for @MikeDvorskiy advice.